### PR TITLE
Migrate deprecated rxjs functions/methods

### PR DESCRIPTION
**Pull request by fkling**
As a follow up to https://github.com/sourcegraph/sourcegraph/pull/61122 this commit updates the deprecated methods with the suggested replacements.

**Note**

This only migrates functions/methods that are replaced with something else, or whose deprecated call signature can easily be identified (e.g. `throwError(error)` -> `throwError(() => error)`). It's possible that there are more functions which deprecate a specific signature that we are using. I'll migrate those as I encounter them.

**Notes about `.toPromise`**

The instances of `.toPromise` converted here are all instances where the updated return value of `Promise<X|undefined>` did not produce a TS error (the ones with errors have been converted in #61122). However that doesn't mean that they can simply be replaced with `firstValueFrom`, `lastValueFrom` (these two methods throw errors when the source observable hasn't emitted a value before closing).
I update the callsites under two assumptions:
- Callsites that involve GraphQL requests will always emit a value and thus can be converted to using `lastValueFrom`/`firstValueFrom`.
- For other callsites we cannot make the assumption that the source observable emits before closing and thus they need a default value.

## Test plan

CI

### CODE REVIEW

**Review by camdencheek — APPROVED** [review]

File: client/common/src/util/rxjs/memoizeObservable.ts [hunk 1]
```diff
@@ -40,8 +40,12 @@ export function memoizeObservable<P, T>(
             return hit
         }
         const observable = func(parameters).pipe(
-            publishReplay(),
-            refCount(),
+            share({
```
Comment (vovakulikov) [inline client/common/src/util/rxjs/memoizeObservable.ts:43]: Same comment about shareReplay as above. 

Here is my sandbox where I tried to compare different operators https://stackblitz.com/edit/eaoqng?devtoolsheight=50&file=index.ts

File: client/browser/src/shared/code-hosts/gerrit/codeHost.ts [hunk 2]
```diff
@@ -355,8 +355,12 @@ export const observeMutations = (
...
         // Wrap in an array, because that's how mutation observers emit events.
         map(mutationRecord => [mutationRecord]),
-        publishReplay(),
-        refCount()
+        share({
+            connector: () => new ReplaySubject(),
+            resetOnError: false,
+            resetOnComplete: false,
+            resetOnRefCountZero: false,
+        })
```
[the remark this answers is not in the document]
Comment (vovakulikov) [reply client/browser/src/shared/code-hosts/gerrit/codeHost.ts:363]: Oh, I missed that we also provide connect which is a replay subject. So it should work as publishReplay at the end, but still not sure why not to use shareReplay instead

**Review by vovakulikov — APPROVED** [review]
No blockers, didn't test it manually in code host integration through

File: client/browser/src/shared/code-hosts/gerrit/codeHost.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fkling) [reply client/browser/src/shared/code-hosts/gerrit/codeHost.ts:363]: @camdencheek, @vovakulikov I was following https://rxjs.dev/deprecations/multicasting#publish . Could it be that `shareReplay` was introduced later on?

**Pull request merged**